features/environment.py

```diff
@@ -412,6 +415,58 @@ def before_scenario(context: Context, scenario: Scenario):
             )
 
 
+def after_scenario(context, scenario):
+    """Collect the coverage files after the scenario is run."""
+    if context.pro_config.collect_coverage:
+        cov_dir = os.path.join(context.pro_config.artifact_dir, "coverage")
+        if not os.path.exists(cov_dir):
+            os.makedirs(cov_dir)
+
+        inner_dir = os.path.basename(scenario.filename.replace(".", "_"))
+        new_artifacts_dir = os.path.join(
+            cov_dir,
+            inner_dir,
+        )
+        if not os.path.exists(new_artifacts_dir):
+            os.makedirs(new_artifacts_dir)
+        if hasattr(context, "machines") and SUT in context.machines:
+            try:
+                scenario_name = (
+                    os.path.basename(scenario.filename.replace(".", "_"))
+                    + "_"
+                    + str(scenario.line)
+                )
+                cov_filename = ".coverage.{}".format(scenario_name)
+                tmp_covfile_path = "/tmp/{cov_filename}".format(
+                    cov_filename=cov_filename
+                )
+                context.machines[SUT].instance.execute(
+                    [
+                        "mv",
+                        "/home/ubuntu/.coverage",
+                        f"{tmp_covfile_path}",
```

Comment:
Let's not use f-strings here because we might run our tests on CI that still don't support it :(